code/js/modules/BaseController.js

```diff
@@ -191,6 +201,18 @@
   };
 
   /**
+   * Returns a bool indicating whether the player's song changed since the last state update
+   * @param {{song: {String}}} newState - new state object
+   * @return {Boolean} true if song just changed, false otherwise
+   */
+  BaseController.prototype.getSongChanged = function(newState) {
+      return this.oldState &&
+            newState &&
+            newState.song &&
```

Comment:
Do you mean "getSongChanged" entirely or just part of the check? Testing now, I remember that I added the check so that notifications would not fire when the play/pause of the track was toggled, and *only* when the song actually changed.